lib/controllers/situations.js

```diff
@@ -68,3 +68,74 @@ exports.openfiscaResponse = function(req, res, next) {
         res.send(result);
     });
 };
+
+function createTestContent(situation, test) {
+    var openfiscaRequest = mapping.buildOpenFiscaRequest(situation, true);
+    var testCase = openfiscaRequest.scenarios[0].test_case;
+    return _.assign(test, {
+        period: openfiscaRequest.scenarios[0].period,
+        absolute_error_margin: 10,
+        individus: testCase.individus,
+        familles: testCase.familles[0],
+        foyers_fiscaux: testCase.foyers_fiscaux[0],
+        menages: testCase.menages[0]
+    });
+}
+
+function myToString(obj) {
+    return obj.toString();
+}
+
+function normalizeTestContent(testFileContent) {
+    testFileContent.individus = testFileContent.individus.map(function(individu) {
+        individu.id = individu.id.toString();
+        return individu;
+    });
+
+    function replaceElementsWithStringRepresentation(obj) {
+        function convert(leafName) {
+            obj[leafName] = obj[leafName].map(myToString);
+        }
+        return convert;
+    }
+
+    var sources = {
+        familles: ['parents', 'enfants'],
+        foyers_fiscaux: ['declarants', 'personnes_a_charge'],
+        menages: ['enfants'],
+    };
+
+    for (var parent in sources) {
+        sources[parent].forEach(replaceElementsWithStringRepresentation(testFileContent[parent]));
+    }
+
+    testFileContent.menages.personne_de_reference = testFileContent.menages.personne_de_reference.toString();
+    if (testFileContent.menages.conjoint) {
+        testFileContent.menages.conjoint = testFileContent.menages.conjoint.toString();
+    }
+}
+
+function createTestInYAML(situation, test) {
+    var testFileContent = createTestContent(situation, test);
+    normalizeTestContent(testFileContent);
+
+    return require('json2yaml').stringify(testFileContent);
+}
+
+exports.openfiscaTest = function(req, res, next) {
+    var test = {
+        name: req.query.name || req.body.name || 'Missing name',
+        description: req.query.description || req.body.description || 'Missing description',
+        output_variables: req.query.expectedResults || req.body.expectedResults,
+    };
+    if (test.output_variables)
+        return res.send(createTestInYAML(req.situation, test));
+
+    // When expected results are not provided use truthy values from OpenFisca simulation.
+    // This could be changed to generate an error.
+    req.situation.simulate(function(err, result) {
+        if (err) return next(err);
+        test.output_variables = _.pickBy(result.calculatedPrestations, function(value) { return Boolean(value); });
```

Comment:
Why do you cast here? Casting something to a `Boolean` simply exposes its truthiness, which is strictly equal to testing the something directly, which is done by `_.pickBy` by default since the default predicate is identity.